server/src/main/java/org/elasticsearch/index/mapper/MapperMetrics.java

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.index.mapper;
+
+/**
+ * Groups together all metrics used in mappers.
+ * Main purpose of this class is to avoid verbosity of passing individual metric instances around.
+ */
+public record MapperMetrics(SourceFieldMetrics sourceFieldMetrics) {
+    public static MapperMetrics NOOP = new MapperMetrics(SourceFieldMetrics.NOOP);
```

Comment:
Are this `NOOP` instance and the other one in `SourceFieldMetrics` only used in non production code? Otherwise maybe we can move it to test framework?

Reply:
Almost - it is used in benchmarks too.

Reply:
I see, maybe then let's keep it here.